cmd/migration-managerd/internal/api/workers.go

```diff
@@ -481,7 +598,13 @@ func (d *Daemon) ensureISOImagesExistInStoragePool(ctx context.Context, instance
 	}
 
 	// Get the target.
-	t, err := d.target.GetByID(ctx, *inst.TargetID)
+	var t migration.Target
+	err = transaction.Do(ctx, func(ctx context.Context) error {
+		var err error
+		t, err = d.target.GetByID(ctx, *inst.TargetID)
+
+		return err
+	})
```

Comment:
Single statement, no need to wrap it in a transaction.

Reply:
This is a good example of what I'm talking about with having a bias for transactions leading to thinking about structuring code better.

You caught that this looked weird. Now as I'm looking at this block more closely, I think it seems wrong to even have db access in this helper. After all, it's just about checking if an Incus target has the right storage volumes. If anything, we should have already instantiated a target client, and this function should expect it as an argument.